rust_snuba/src/processors/generic_metrics.rs

```diff
@@ -89,16 +90,61 @@ enum MetricValue {
     },
 }
 
+trait Decodable<const SIZE: usize>: Copy {
+    const SIZE: usize = SIZE;
+
+    fn decode_bytes(bytes: [u8; SIZE]) -> Self;
+}
+
+impl Decodable<4> for u32 {
+    fn decode_bytes(bytes: [u8; Self::SIZE]) -> Self {
+        Self::from_le_bytes(bytes)
+    }
```

Comment:
Slightly unrelated question to this pr, but I was hoping @jjbayer or @Dav1dde can answer this. Why does `from_le_bytes` not take a slice but requires the full ownership of the `[u8]`? Is it because it is trivially copyable (not sure how this concept maps to rust)? It seems like with the `try_into` on line 136 we are effectively performing a `memcpy` of the bytes array.

Reply:
The few bytes that make up an integer or a floating point number need to be copied out of the slice at some point anyway. As for `try_into` itself, I do hope that the compiler optimizes out any unnecessary copies. You can check https://godbolt.org/ to be sure.

There still might be a possible optimization where we reinterpret the entire array of bytes as an array of numbers without copying anything (relying on the assumption that the platform represents numbers in little endian notation). But I wouldn't go there in the first iteration.

Reply:
> There still might be a possible optimization where we reinterpret the entire array of bytes as an array of numbers without copying anything (relying on the assumption that the platform represents numbers in little endian notation). But I wouldn't go there in the first iteration.

Would that be with `std::mem::transmute`?

Reply:
Yes, for example

```rust
unsafe {
    std::mem::transmute::<&[u8], &[u32]>(&raw_bytes)
}
```